src/codegen/compile.ml

```diff
@@ -7813,7 +7840,7 @@ module Serialization = struct
         E.else_trap_with env ("IDL error: left-over bytes " ^ ts_name) ^^
         ReadBuf.is_empty env get_ref_buf ^^
         E.else_trap_with env ("IDL error: left-over references " ^ ts_name)
```

Comment:
TODO: maybe reset transient Registers to `0` on exit to avoid GC holes.

Reply:
Thanks for reminding. I added the corresponding safety check.